src/hardware/mixer.cpp

```diff
@@ -1293,6 +1298,38 @@ void MixerChannel::SetLowPassFilter(const FilterState state)
 	}
 }
 
+void MixerChannel::ConfigureNoiseGate(const float threshold_db, const float attack_time_ms,
+                                      const float release_time_ms)
+{
+	assert(attack_time_ms > 0.0f);
+	assert(release_time_ms > 0.0f);
+
+	noise_gate.threshold_db    = threshold_db;
+	noise_gate.attack_time_ms  = attack_time_ms;
+	noise_gate.release_time_ms = release_time_ms;
+
+	InitNoiseGate();
+}
+
+void MixerChannel::EnableNoiseGate(const bool enabled)
+{
+	LOG_MSG("%s: Noise gate %s", name.c_str(), (enabled ? "enabled" : "disabled"));
+	do_noise_gate = enabled;
```

Comment:
Same comment here. It's currently fine. Since this is just a bool, you can make it `std::atomic` instead of using a mutex to allow for future runtime changes.

Reply:
Actually a mutex would probably be faster since this function is rarely called. That way the mixer can read this variable non-atomically since it acquires the mutex first anyway. Very unlikely to cause a measurable difference either way though.

Reply:
I've added the lock guard to all public MixerChannel methods, as discussed.